x-pack/plugins/inference/server/chat_complete/adapters/gemini/process_vertex_stream.ts

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { Observable } from 'rxjs';
+import {
+  ChatCompletionChunkEvent,
+  ChatCompletionTokenCountEvent,
+  ChatCompletionEventType,
+} from '../../../../common/chat_complete';
+import type { GenerateContentResponseChunk } from './types';
+
+export function processVertexStream() {
+  return (source: Observable<GenerateContentResponseChunk>) =>
+    new Observable<ChatCompletionChunkEvent | ChatCompletionTokenCountEvent>((subscriber) => {
+      function handleNext(value: GenerateContentResponseChunk) {
+        // completion: only present on last chunk
+        if (value.usageMetadata) {
+          subscriber.next({
+            type: ChatCompletionEventType.ChatCompletionTokenCount,
+            tokens: {
+              prompt: value.usageMetadata.promptTokenCount,
+              completion: value.usageMetadata.candidatesTokenCount,
+              total: value.usageMetadata.totalTokenCount,
+            },
+          });
+        }
+
+        const contentPart = value.candidates![0].content.parts[0];
```

Comment:
should we change `!` to `?` given we already do an non-nullish check on line 36?